openassessment/xblock/static/js/src/lms/oa_base.js

```diff
@@ -73,6 +76,25 @@ OpenAssessment.BaseView.prototype = {
     },
 
     /**
+     * Install click handlers to set usage key of an XBlock.
+     */
+    setUpUsageID: function() {
+        $(".action--submit").on("click", function(event) {
+            usageID = $(event.target).closest('.xblock-student_view-openassessment').data('usage-id');
+        });
+    },
+
+    /**
+     * Get usage key of an XBlock.
+     */
+    getUsageID: function() {
+        if (!usageID) {
+            usageID = $(this.element).data('usage-id');
```

Comment:
Normally, I would not have need this, I could have got the usage id directly from the peer/self-assessment view as  `this.baseview.usageID` but sometimes (user has submitted some steps and refreshes or visits after browsing to the ORA problem) the usageID would not be set. So I had created this utility methods `getUsageId` that would always give us correct usage ID.